### ISSUE TITLE
Instructions for hosting on github are not accurate and fail to deploy with script. 

### ISSUE THREAD

**Issue by adrianmoya** [spf13/hugo#1777]
Hi, I'm following the "Hosting Personal/Organization Pages" instructions on the docs. When trying to deploy using the deploy.sh script, I was not getting the desired result. Reading on the web I found out that I had to add the submodule using the `-b master` parameter so that the submodule tracked the correct branch and the deploy.sh script work. Otherwise it stays in a detached state which doesn't work.

So step 6 should be:

`git submodule add -b master git@github.com:<username>/<username>.github.io.git public`

I'm using git version 1.9.1 (in case that an updated version of git handles this differently)

**Comment by anthonyfok** [comment]
Thanks @adrianmoya!  Should be live soon.

**Comment by adrianmoya** [comment]
@anthonyfok Hey Anthony, it looks like this is not enought to prevent git from complaining when trying to deploy. At least for me, that I use two different machines. If you don't pull the latest published version from github, the push will fail. 

I came up with a few enhacements for the deploy script, please check them out and see if this could help others on the community. I know this is an issue totally git related, but it's a common scenario to use github as hosting provider.

https://github.com/adrianmoya/adrianmoya.com-src/blob/master/deploy.sh

Basically, the changes are: 
1) If commits are pending, exit and tell the user to commit changes
2) Update submodule public to latest reference
3) Commit and pull as usual (the current script)
4) I refresh the reference on the source repo. 

This way, when updating the project on the second machine, and trying to deploy, it will pull the latest version of the live site and there will be no rejected push. I'm still testing the script, feedback welcome!

The docs could benefit from a "Seting up another machine" section.

**Comment by bep** [comment]
This issue has been automatically marked as stale because it has not been commented on for at least four months.

The resources of the Hugo team are limited, and so we are asking for your help.

If this is a **bug** and you can still reproduce this error on the <code>master</code> branch, please reply with all of the information you have about it in order to keep the issue open.

If this is a **feature request**, and you feel that it is still valuable, please open a proposal at https://discuss.gohugo.io/.

This issue will automatically be closed in four months if no further activity occurs. Thank you for all your contributions.

**Comment by bep** [comment]
**Note/Update:** This issue is marked as stale, and I may have said something earlier about "opening a thread on the discussion forum". Please don't.

If this is a **bug** and you can still reproduce this error on the latest <code>release</code> or the <code>master</code> branch, please reply with all of the information you have about it in order to keep the issue open.

If this is a **feature request**, and you feel that it is still relevant and valuable, please tell us why.

**Comment by rdwatters** [comment]
This issue is obsolete with the new docs, which include a revised version of the GH hosting.
